jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java

```diff
@@ -0,0 +1,394 @@
+                    builder = builder.field(field, Collections.emptyList());
+                }
+
+            }
+
+            builder = builder.endObject();
+            IndexRequest indexRequest = new IndexRequest(indexName, docDef.getEntityField(), entity.getId())
+                    .source(builder);
+
+            String addUpdateScript = "if(ctx._source.<fieldName> == null || ctx._source.<fieldName>.empty) " +
+                    "{ctx._source.<fieldName>=['<fieldValue>'] } else {ctx._source.<fieldName>.add('<fieldValue>')}";
+            addUpdateScript = addUpdateScript.replaceAll("<fieldName>", fieldToAdd).replaceAll("<fieldValue>", fieldValueToAdd);
+
+            UpdateRequest upReq = new UpdateRequest(indexName, docDef.getEntityField(), entity.getId())
+                    .script(new Script(addUpdateScript))
+                    .upsert(indexRequest);
+
+            UpdateResponse response = client.update(upReq).get();
+
+            LOGGER.debug("Received the following Update response : " + response + " for the following entity: " + entity);
+
+        } catch(Exception e) {
+            throw new TextIndexException("Unable to Index the Entity in ElasticSearch.", e);
+        }
+    }
+
+    /**
+     * Delete an entity.
+     * Since we are storing different predicate values within the same indexed document,
+     * deleting the document using entity Id is sufficient to delete all the related contents for a given entity.
+     * @param entity entity to delete
+     */
+    @Override
+    public void deleteEntity(Entity entity) {
+
+        String fieldToRemove = null;
+        String valueToRemove = null;
+        for(String field : docDef.fields()) {
+            if(entity.get(field) != null) {
+                fieldToRemove = field;
+                valueToRemove = (String)entity.get(field);
+                break;
+            }
+        }
+
+        String script = "if(ctx._source.<fieldToRemove> != null && (ctx._source.<fieldToRemove>.empty != true) " +
+                "&& (ctx._source.<fieldToRemove>.indexOf('<valueToRemove>') >= 0)) " +
+                "{ctx._source.<fieldToRemove>.remove(ctx._source.<fieldToRemove>.indexOf('<valueToRemove>'))}";
+        script = script.replaceAll("<fieldToRemove>", fieldToRemove).replaceAll("<valueToRemove>", valueToRemove);
+
+        UpdateRequest updateRequest = new UpdateRequest(indexName, docDef.getEntityField(), entity.getId())
+                .script(new Script(script));
+
+        try {
+            client.update(updateRequest).get();
+        }catch(Exception e) {
+            throw new TextIndexException("Unable to delete entity.", e);
+        }
+
+        LOGGER.debug("deleting content related to entity: " + entity.getId());
+
+    }
+
+    /**
+     * Get an Entity given the subject Id
+     * @param uri the subject Id of the entity
+     * @return a map of field name and field values;
+     */
+    @Override
+    public Map<String, Node> get(String uri) {
```

Comment:
As discussed already, this method may go away. For now, just don't waste any more time on it.